### ISSUE TITLE
Docs: [consistent-type-imports] Revise recommended usage

### ISSUE THREAD

**Issue by aaronadamsCA** [typescript-eslint/typescript-eslint#11681]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [x] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

The `consistent-type-imports` page specifically recommends that if you use `verbatimModuleSyntax`, you should turn off this rule:

> Because there are some differences, using both this rule and `verbatimModuleSyntax` at the same time can lead to conflicting errors. As such we recommend that you only ever use one _or_ the other -- never both.

But this rule does something very important that `verbatimModuleSyntax` does not: it makes sure you use `import type` for imports used only as types.

> `consistent-type-imports` is concerned with exactly one thing - ensuring your value imports are used as values, warning you when your value imports aren't used as values. https://github.com/typescript-eslint/typescript-eslint/issues/6379#issuecomment-1405902005

So in this situation:

```ts
import { GraphQLSchema } from "graphql";

export class Foo {
  private readonly schema: GraphQLSchema;

  constructor(schema: GraphQLSchema) {
    this.schema = schema;
  }

  // …
}
```

If you enable `verbatimModuleSyntax`, oops, now you have a side-effect import of `"graphql"`—and all of the weird problems that brings (https://github.com/graphql/graphql-js/issues/2801).

For this reason, I think it would be good to update the recommendations for this rule, since it can specifically help guard against unwanted side effects once `verbatimModuleSyntax` is enabled.

### Affected URL(s)

https://typescript-eslint.io/rules/consistent-type-imports

**Comment by kirkwaiblinger** [comment]
Yeah, this does seem like a gap... also closely related is [no-import-type-side-effects](https://typescript-eslint.io/rules/no-import-type-side-effects), but that only flags `import { type X } from 'foo'` 🤔 

@typescript-eslint/triage-team

**Comment by kirkwaiblinger** [comment]
I'll note that `verbatimModuleSyntax` seems to be picking up traction, so I'm +1 to making sure it's well supported here.

**Comment by kyungilcho** [comment]
I'd like to work on this if it's still available. I'll take a look at revising the `consistent-type-imports` docs around `verbatimModuleSyntax` and propose a concrete wording update.
